### ISSUE TITLE
Generate "default objects"

### ISSUE THREAD

**Issue by sdboyer** [sdboyer/cuetsy#6]
### Problem

Currently, we disregard CUE default values. This needs to change, though there's some flexibility here on exactly how we do it.

### Proposed Solution

@ryantxu can say more about exactly how he thinks this ought to be shaped, but...

We should optionally generate:

* A `const` with default values assigned for the type
* A default-setter func:`<typename>setDefaults(t <typename>)` 
* An `*isDefault()` func that checks whether an instance of the type equal has the default values.

Simple example:

```cue
Foo: {
  Bar: string | *"ohai*
  Baz: int | *4
} @cuetsy(targetType="interface")
```

```typescript
export interface Foo {
  Bar: string;
  Baz: int;
}

export const FooDefault {
  Bar: 'ohai';
  Baz: 4
}

export function FooIsDefault(obj: Foo): bool {
  ...
}

export function FooSetDefaults(obj: Foo): Foo {
  ...
}
```

i feel like there's probably a more elegant generic way of expressing the funcs, maybe even the default, perhaps leveraging some [Partials](https://www.typescriptlang.org/docs/handbook/utility-types.html#partialtype). 🤷 

This will be tricky. For one, i think we'll have to read through CUE references in a way we don't now. e.g.:

```cue
AType: "foo" | "bar" | *"baz" @cuetsy(targetType="type")
BType: {
  Composed: AType
} @cuetsy(targetType="interface")
```

For `BType`, the exported const would need to traverse the reference to `AType` to figure out the right default value for the exported const's `Composed` field.

Also, related, we can't assume that even required fields will have a default. I believe that means that the exported const isn't actually guaranteed to satisfy the exported interface. (That may just be more of an oddity than problematic, though).

In general, though, we don't need this to cover all possible CUE constructions perfectly, as an operating assumption in cuetsy, at least for now, is that folks are going to be checking the generated TS on the other side, and they can choose to rewrite their CUE if it doesn't come out right. (Yup, we iz prototyping!)

**Comment by ryantxu** [comment]
I don't see any need for cuetsy to worry about the functions:
```
export function FooIsDefault(obj: Foo): bool {
  ...
}

export function FooSetDefaults(obj: Foo): Foo {
  ...
}
```

This can be solved using pretty standard strategies like:
* https://lodash.com/docs/4.17.15#defaultsDeep
* https://lodash.com/docs/4.17.15#isEqual

The important part for cuetsy is to generate the const object:
```
export const fooDefault {
  Bar: 'ohai';
  Baz: 4
}
```

**Comment by sdboyer** [comment]
Fixed by #7
